File: ex/uri1110.cpp
#include <iostream>
#include <cstdlib>
#include <vector>
using namespace std;

int* populate_deck(int deck_size)
{
	int* deck = (int*) malloc(sizeof(int)*deck_size);

	for (int i = 0; i < deck_size; ++i)
		deck[i] = i + 1;

	return deck;
}

main()
{
	int deck_size, temp;
	int* deck;
	vector<int> discard;

	cin >> deck_size;
	while (deck_size > 0)
	{
		deck = populate_deck(deck_size);
		while (deck_size > 1)
		{
			discard.push_back(deck[0]);
			temp = deck[1];
			for (int i = 0; i < deck_size - 3; ++i) 
				deck[i] = deck[i+2];
			deck[deck_size-2] = temp;
			deck_size--;
		}

		cout << "Discarded cards:";
		for (vector<int>::iterator it = discard.begin(); it != discard.end(); it++)
			cout << " " << *it;
		cout << endl;
		cout << "Remaining cards: " << deck[0] << endl;
		
		discard.clear();
		cin >> deck_size;
	}
}

File: ex/uri1110.java
import java.io.InputStreamReader;
import java.io.BufferedReader;
import java.io.IOException;
import java.util.List;
import java.util.ArrayList;

public class uri1110 {
  static InputStreamReader ir = new InputStreamReader(System.in);
  static BufferedReader in = new BufferedReader(ir);

  static List populateDeck(int size) {
    List sequence = new ArrayList<Integer>();
    for (int n = 1; n <= size; ++n) sequence.add(n);
    return sequence;
  }

  public static void main(String[] args) throws IOException {
    int deckSize, c;
    List deck, discardedCards;

    deckSize = Integer.parseInt(in.readLine());
    while (deckSize > 0) {
      deck = populateDeck(deckSize);
      discardedCards = new ArrayList<Integer>();

      while (deck.size() > 1) {
        discardedCards.add(deck.remove(0));
        deck.add(deck.remove(0));
      }

      System.out.print("Discarded cards: ");
      for (c = 0; c < discardedCards.size() - 1; c++)
        System.out.print(discardedCards.get(c) + ", ");
      System.out.println((int) discardedCards.get(c));

      System.out.print("Remaining card: ");
      System.out.println((int) deck.get(0));

      deckSize = Integer.parseInt(in.readLine());
    }
  }

}
